### PR TITLE
Fix #5799 - swagger for mergePullRequest

### LINKED ISSUE #5799: repoMergePullRequest Swagger  not up-to-date
- Gitea version (or commit ref): eb0ee6b
- Git version:
- Operating system:
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes https://try.gitea.io/api/swagger#/repository/repoMergePullRequest
  - [ ] No
  - [ ] Not relevant

## Description
Swagger description doesn't reflect API for repoMergePullRequest endpoint.
Some parameters needs to be sent: https://github.com/go-gitea/gitea/blob/master/modules/auth/repo_form.go#L366
```go
type MergePullRequestForm struct {
	Do                string `binding:"Required;In(merge,rebase,rebase-merge,squash)"`
	MergeTitleField   string
	MergeMessageField string
}
```
But the swagger description doesn't mention them: https://github.com/go-gitea/gitea/blob/master/templates/swagger/v1_json.tmpl#L3660
```json
...
{
        "produces": [
          "application/json"
        ],
        "tags": [
          "repository"
        ],
        "summary": "Merge a pull request",
        "operationId": "repoMergePullRequest",
        "parameters": [
          {
            "type": "string",
            "description": "owner of the repo",
            "name": "owner",
            "in": "path",
            "required": true
          },
          {
            "type": "string",
            "description": "name of the repo",
            "name": "repo",
            "in": "path",
            "required": true
          },
          {
            "type": "integer",
            "format": "int64",
            "description": "index of the pull request to merge",
            "name": "index",
            "in": "path",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "$ref": "#/responses/empty"
          },
          "405": {
            "$ref": "#/responses/empty"
          }
        }
      }
...
}
```

### PR DESCRIPTION
**Pull request by xabufr**
Fixes https://github.com/go-gitea/gitea/issues/5799

Add missing definitions in swagger for merge pull request API.

### CODE REVIEW

**Comment by sapk** [comment]
@xabufr the swagger .json file is a generated file. 
You need to define the struct in comments of the go struct. Like https://github.com/go-gitea/gitea/blob/master/routers/api/v1/swagger/user.go#L33
And use the command `make generate-swagger` to generate the json file.

**Comment by xabufr** [comment]
Indeed, I didn't see. Is it good now ?

**Review by sapk — APPROVED** [review]
Thanks for the contribution.

**Review by zeripath — APPROVED** [review]

File: modules/auth/repo_form.go [hunk 1]
```diff
@@ -364,7 +364,10 @@ func (f *InitializeLabelsForm) Validate(ctx *macaron.Context, errs binding.Error
 //                                     \/     \/   |__|           \/     \/
 
 // MergePullRequestForm form for merging Pull Request
+// swagger:model MergePullRequestOption
 type MergePullRequestForm struct {
+	// required: true
+	// enum: merge, rebase, rebase-merge, squash
```
Comment (lafriks) [inline modules/auth/repo_form.go:370]: This should have been without spaces to have correct values in swagger json

**Pull request merged**